src/components/ScrollView.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import type {ForwardedRef} from 'react';
+// eslint-disable-next-line no-restricted-imports
+import {ScrollView as RNScrollView} from 'react-native';
+import type {ScrollViewProps as RNScrollViewProps} from 'react-native';
+
+type ScrollViewProps = RNScrollViewProps & {
+    /** whether to add right inset from the edges of the scroll view on iOS */
+    shouldAddRightInsetForIndicator?: boolean;
```

Comment:
I doubt whether this prop is required or not bcoz we are prioritizing the `scrollIndicatorInsets` value and in case it is not available we should consider this right inset to min number as a default for iOS.